src/sentry/tasks/statistical_detectors.py

```diff
@@ -113,30 +113,24 @@ def run_detection() -> None:
     performance_projects_count = 0
     profiling_projects_count = 0
 
-    for project in RangeQuerySetWrapper(
-        Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
-        step=100,
+    for project_id, flags in RangeQuerySetWrapper(
+        Project.objects.filter(status=ObjectStatus.ACTIVE).values_list("id", "flags"),
+        result_value_getter=lambda item: item[0],
     ):
-        if project.flags.has_transactions and (
-            features.has(
-                "organizations:performance-statistical-detectors-ema", project.organization
-            )
-        ):
-            performance_projects.append(project)
+        if flags & Project.flags.has_transactions:
+            performance_projects.append(project_id)
             performance_projects_count += 1
 
             if len(performance_projects) >= PROJECTS_PER_BATCH:
                 detect_transaction_trends.delay(
-                    list({p.organization_id for p in performance_projects}),
-                    [p.id for p in performance_projects],
+                    [],
```

Comment:
We no longer require the task to be passed a list of org ids as that will be fetched in the task.